src/common/misc/FileFunctions.C

```diff
@@ -647,17 +652,50 @@ basename(char const *path, int& start, char const *suffix=0)
    }
 }
 
+// **************************************************************************
+//  Modifications:
+//    Kathleen Biagas, Thu Nov 12, 2020
+//    Use std::filesystem::path on Windows, it does the right thing
+//    regardless of the type of path separator used.
```

Comment:
Our funcs should work regardless of separator too. If they don't shouldn't we just fix that? Reason I ask is that it feels more complicated to support two wholly different approaches on two different platforms. Since we already have home-grown methods on Linux, I think we get more consistent behavior and less complex coding if we make that one way work every where. Don't you agree?

Reply:
Actually, I don't see an issue having different code paths. The assumptions for 'basename' are unix-basename specific (see your own comments on the method).  On Windows,  paths ending with  a path-separator are considered only as 'path' (or directory in windows-parlance), with no filename attached, but current basename logic says that 'C:\usr\' should return 'usr' as the basename.   And 'C:' is not a path, it is a root name, the path is 'C:\', so Dirname currently returns the wrong thing because it strips off the trailing slash when passed 'C:'
I would need to write windows-specific logic into both of these methods, so why not simplify matters and use something that already works correctly?